Parser throws CCE for incomplete worker sync send

**Description:**
Consider the following source,

```
public function main() {
    worker w1 {
        int i = 100;
        () sendVal = i ->> 
    }
    
    worker w2 {
        int vW1 = 0;
        vW1 = <- w1;
    }
}

```

Parser throws the following class cast exception

```
Caused by: java.lang.ClassCastException: io.ballerinalang.compiler.syntax.tree.IdentifierToken cannot be cast to io.ballerinalang.compiler.syntax.tree.SimpleNameReferenceNode
	at io.ballerinalang.compiler.syntax.tree.SyncSendActionNode.peerWorker(SyncSendActionNode.java:44)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:2007)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:410)
	at io.ballerinalang.compiler.syntax.tree.SyncSendActionNode.apply(SyncSendActionNode.java:54)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createActionOrExpression(BLangNodeTransformer.java:3738)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createExpression(BLangNodeTransformer.java:3707)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createBLangVarDef(BLangNodeTransformer.java:2400)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:2369)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:410)
	at io.ballerinalang.compiler.syntax.tree.VariableDeclarationNode.apply(VariableDeclarationNode.java:67)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.generateAndAddBLangStatements(BLangNodeTransformer.java:3618)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.generateBLangStatements(BLangNodeTransformer.java:3606)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:2320)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:410)
	at io.ballerinalang.compiler.syntax.tree.BlockStatementNode.apply(BlockStatementNode.java:54)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:1336)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:410)
	at io.ballerinalang.compiler.syntax.tree.NamedWorkerDeclarationNode.apply(NamedWorkerDeclarationNode.java:63)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:1285)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:410)
	at io.ballerinalang.compiler.syntax.tree.FunctionBodyBlockNode.apply(FunctionBodyBlockNode.java:59)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createFunctionNode(BLangNodeTransformer.java:1174)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:1132)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:410)
	at io.ballerinalang.compiler.syntax.tree.FunctionDefinitionNode.apply(FunctionDefinitionNode.java:71)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:477)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:410)
	at io.ballerinalang.compiler.syntax.tree.ModulePartNode.apply(ModulePartNode.java:54)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.accept(BLangNodeTransformer.java:435)
	at org.wso2.ballerinalang.compiler.parser.Parser.generateCompilationUnitNew(Parser.java:143)
	at org.wso2.ballerinalang.compiler.parser.Parser.parseNew(Parser.java:116)
	at org.wso2.ballerinalang.compiler.PackageLoader.parse(PackageLoader.java:500)
	at org.wso2.ballerinalang.compiler.PackageLoader.loadEntryPackage(PackageLoader.java:380)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:156)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:137)
	at org.ballerinalang.langserver.compiler.LSModuleCompiler.compilePackagesSafe(LSModuleCompiler.java:291)
	... 11 more
```

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**
<!-- Any related issues such as sub tasks, issues reported in other repositories (e.g component repositories), similar problems, etc. -->

**Suggested Labels (optional):**
<!-- Optional comma separated list of suggested labels. Non committers can’t assign labels to issues, so this will help issue creators who are not a committer to suggest possible labels-->

**Suggested Assignees (optional):**
<!--Optional comma separated list of suggested team members who should attend the issue. Non committers can’t assign issues to assignees, so this will help issue creators who are not a committer to suggest possible assignees-->

--- a/compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java
+++ b/compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java
@@ -10842,7 +10842,11 @@ public class BallerinaParser extends AbstractParser {
                 return STNodeFactory.createSimpleNameReferenceNode(consume());
             default:
                 Solution sol = recover(token, ParserRuleContext.PEER_WORKER_NAME);
-                return sol.recoveredNode;
+                if (sol.action == Action.REMOVE) {
+                    return sol.recoveredNode;
+                }
+
+                return STNodeFactory.createSimpleNameReferenceNode(sol.recoveredNode);
         }
     }
 

--- a/compiler/ballerina-parser/src/test/resources/actions/receive-action/receive_action_assert_02.json
+++ b/compiler/ballerina-parser/src/test/resources/actions/receive-action/receive_action_assert_02.json
@@ -80,11 +80,17 @@
                       ]
                     },
                     {
-                      "kind": "IDENTIFIER_TOKEN",
-                      "isMissing": true,
+                      "kind": "SIMPLE_NAME_REFERENCE",
                       "hasDiagnostics": true,
-                      "diagnostics": [
-                        "ERROR_MISSING_IDENTIFIER"
+                      "children": [
+                        {
+                          "kind": "IDENTIFIER_TOKEN",
+                          "isMissing": true,
+                          "hasDiagnostics": true,
+                          "diagnostics": [
+                            "ERROR_MISSING_IDENTIFIER"
+                          ]
+                        }
                       ]
                     }
                   ]
@@ -142,11 +148,17 @@
                       "kind": "LEFT_ARROW_TOKEN"
                     },
                     {
-                      "kind": "IDENTIFIER_TOKEN",
-                      "isMissing": true,
+                      "kind": "SIMPLE_NAME_REFERENCE",
                       "hasDiagnostics": true,
-                      "diagnostics": [
-                        "ERROR_MISSING_IDENTIFIER"
+                      "children": [
+                        {
+                          "kind": "IDENTIFIER_TOKEN",
+                          "isMissing": true,
+                          "hasDiagnostics": true,
+                          "diagnostics": [
+                            "ERROR_MISSING_IDENTIFIER"
+                          ]
+                        }
                       ]
                     }
                   ]
@@ -480,11 +492,17 @@
                                   "kind": "COLON_TOKEN"
                                 },
                                 {
-                                  "kind": "IDENTIFIER_TOKEN",
-                                  "isMissing": true,
+                                  "kind": "SIMPLE_NAME_REFERENCE",
                                   "hasDiagnostics": true,
-                                  "diagnostics": [
-                                    "ERROR_MISSING_IDENTIFIER"
+                                  "children": [
+                                    {
+                                      "kind": "IDENTIFIER_TOKEN",
+                                      "isMissing": true,
+                                      "hasDiagnostics": true,
+                                      "diagnostics": [
+                                        "ERROR_MISSING_IDENTIFIER"
+                                      ]
+                                    }
                                   ]
                                 }
                               ]

--- a/compiler/ballerina-parser/src/test/resources/actions/send-action/send_action_assert_04.json
+++ b/compiler/ballerina-parser/src/test/resources/actions/send-action/send_action_assert_04.json
@@ -101,11 +101,17 @@
                       ]
                     },
                     {
-                      "kind": "IDENTIFIER_TOKEN",
-                      "isMissing": true,
+                      "kind": "SIMPLE_NAME_REFERENCE",
                       "hasDiagnostics": true,
-                      "diagnostics": [
-                        "ERROR_MISSING_IDENTIFIER"
+                      "children": [
+                        {
+                          "kind": "IDENTIFIER_TOKEN",
+                          "isMissing": true,
+                          "hasDiagnostics": true,
+                          "diagnostics": [
+                            "ERROR_MISSING_IDENTIFIER"
+                          ]
+                        }
                       ]
                     }
                   ]
@@ -184,11 +190,17 @@
                       ]
                     },
                     {
-                      "kind": "IDENTIFIER_TOKEN",
-                      "isMissing": true,
+                      "kind": "SIMPLE_NAME_REFERENCE",
                       "hasDiagnostics": true,
-                      "diagnostics": [
-                        "ERROR_MISSING_IDENTIFIER"
+                      "children": [
+                        {
+                          "kind": "IDENTIFIER_TOKEN",
+                          "isMissing": true,
+                          "hasDiagnostics": true,
+                          "diagnostics": [
+                            "ERROR_MISSING_IDENTIFIER"
+                          ]
+                        }
                       ]
                     }
                   ]

